### PR TITLE
Bump tensorflow to 2.16.1 with tf-keras in CI

### PR DESCRIPTION
**Pull request by lillian542**
Updates tests to run with the most recent version of Tensorflow, using the new `tf-keras` package to continue running with Keras 2 (otherwise Keras 3 is used, which is not yet compatible with Pennylane)

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-58874]

File: .github/workflows/install_deps/action.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ inputs:
   tensorflow_version:
     description: The version of TensorFlow to install for any job that requires TensorFlow
     required: false
-    default: 2.15.0
+    default: 2.16.1
```
Comment (lillian542) [inline .github/workflows/install_deps/action.yml:26]:
```suggestion
    default: 2.16
```

**Comment by lillian542** [comment]
[sc-58874]

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks again!

**Comment by PietropaoloFrisoni** [comment]
My understanding is that [these changes](https://github.com/PennyLaneAI/pennylane/pull/5488) should appear here (which is why the changelog has not been updated in this PR).

**Comment by lillian542** [comment]
> My understanding is that https://github.com/PennyLaneAI/pennylane/pull/5488 should appear here (which is why the changelog has not been updated in this PR).

Here we just changed the workflow files for CI tests to install the newest version of Tensorflow. I'd say we generally don't update the changelog when we just bump the versions in tests, if no change to the code is needed, since nothing in the code changed 😅 Sometimes when we bump, we also have to make changes to get the tests to pass, in which case we would add something in the changelog about "updating for TensorFlow XX". But In this case, there was nothing to update, everything passes with the version we pin to here.

**Review by mudit2812 — APPROVED** [review]

**Pull request merged**